src/routes/docs/products/functions/develop-locally/+page.markdoc

```diff
@@ -0,0 +1,88 @@
+---
+layout: article
+title: Develop locally
+description: Learn to develop Appwrite functions locally.
+---
+
+Develop your Appwrite functions locally to make code changes without redeploying your function on every code change and hot reload your code for faster testing.
+
+# Setup {% #setup %}
+
+We use Docker to replicate the production environment for the local deployment of functions. These can be executed locally with the CLI command, which requires initializing a project with an `appwrite.json` file and having local code to run the function locally. The CLI also supports various other [CLI commands](/docs/tooling/command-line/commands).
+
+1. Install the [Docker CLI](https://www.docker.com/products/docker-desktop/)
+2. Ensure Docker is running in the background
+3. Install the [Appwrite CLI](/docs/tooling/command-line/installation#installation)
+4. [Log in](/docs/tooling/command-line/installation#get-started) to your Appwrite account using `appwrite login`
+5. [Initialize your project](/docs/tooling/command-line/installation#initialization)
+6. [Initialize an Appwrite function](/docs/tooling/command-line/functions) and copy and paste your code
+
+# Develop {% #develop %}
+
+Use the `appwrite run functions` command to develop your function locally.
+
+```sh
+appwrite run functions
+? Which function would you like to develop locally? My Awesome Function ("<FUNCTION_ID>")
+
+ runtime   | entrypoint  | path                           | commands     
+-----------|-------------|--------------------------------|--------------
+ node-16.0 | src/main.js | functions/\<FUNCTION_ID\>      | npm install  
+
+ℹ Info: If you wish to change your local settings, update the appwrite.json file and rerun the 'appwrite run' command.
+♥ Hint: Permissions, events, CRON and timeouts dont apply when running locally.
+ℹ Info: Pulling Docker image ...
+♥ Hint: This may take a few minutes, but we only need to do this once.
+ℹ Info: Building function using Docker ...
+Preparing for build ...
+
+Building ...
+
+
+added 4 packages, and audited 5 packages in 2s
+
+
+1 package is looking for funding
+  run `npm fund` for details
+
+
+found 0 vulnerabilities
+
+Packing build ...
+
+Build finished.
+
+ℹ Info: Starting function using Docker ...
+♥ Hint: Function automatically restarts when you edit your code.
+✓ Success: Visit http://localhost:3000/ to execute your function.
+```
+
+This command helps you efficiently develop your Appwrite functions on your local machine. When developing your Appwrite function locally, it will receive [headers](/docs/products/functions/develop#headers) like when you develop a function and it is deployed to Appwrite.
```

Comment:
It's still important to mention the function will update when you modify the code.